java/arcs/core/analysis/RecipeGraph.kt

```diff
@@ -163,4 +120,50 @@ class RecipeGraph(recipe: Recipe) {
             override fun toString() = "[$debugName]"
         }
     }
+
+    companion object {
+        /** Factory method to convert a [Recipe] into a [RecipeGraph]. */
+        operator fun invoke(recipe: Recipe): RecipeGraph {
```

Comment:
are you confident about this?
```
val recipe = Recipe(...)
val graph = recipe()
```
?

Reply:
This method is on the companion object (i.e. it's a static method). It's basically another constructor for RecipeGraph:

```
val graph = RecipeGraph(recipe)
```

(I would have made it a secondary constructor, but there's an unusual requirement that you immediately call `this()`, and I had a lot of processing I had to do to compute the arguments to pass to `this()`)